processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryConfig.java

```diff
@@ -102,6 +107,9 @@
   // Max on-disk temporary storage, per-query; when exceeded, the query fails
   private long maxOnDiskStorage = 0L;
 
+  @JsonProperty
+  private long defaultOnDiskStorage = Integer.MAX_VALUE;
```

Comment:
Both `maxOnDiskStorage` and this new config can be changed to type of `HumanReadableBytes`